addons/l10n_de/migrations/2.0/pre-migrate.py

```diff
@@ -12,6 +12,35 @@ def rename_tag(cr, old_tag, new_tag):
 
 
 def migrate(cr, version):
+    # By deleting tag B from ir_model_data we ensure that the ORM won't try to remove this record.
+    # This is done because the tag might be already used as a FK somewhere else.
+    cr.execute(
+        """DELETE FROM ir_model_data
+           WHERE module='l10n_de'
+           AND name='tag_de_liabilities_bs_B'
+        """
+    )
+
+    # As some people already upgraded, they will have renamed the C_1 tag to B_1. This doesn't come from the script but from the
+    # account_account_tags_data.xml. If they try to run this script now in the fix they get the error that B_1 already exists.
+    # To fix this we can check if it exists or not, and if it does then we don't run the script. This means that the ones
+    # that upgraded won't have the old tags data transferred to the new tags but they will still be able to have the updated sheet.
+    cr.execute(
+        """SELECT 1 FROM ir_model_data
+           WHERE module='l10n_de' AND name='tag_de_liabilities_bs_B_1'
+        """)
```

Comment:
I'm wondering if it's not better to use the orm for that :thinking: 
```py
    env = api.Environment(cr, SUPERUSER_ID, {})
   is_updated = env.ref('l10n_de.tag_de_liabilities_bs_B_1', raise_if_not_found=True)
```
@KangOl What do you think? It is bringing an extra cost to instantiate a new environement there?